packages/openapi-fetch/src/index.ts

```diff
@@ -51,27 +51,21 @@ export interface DefaultParamsOption {
   params?: { query?: Record<string, unknown> };
 }
 
-export interface EmptyParameters {
-  query?: never;
-  header?: never;
-  path?: never;
-  cookie?: never;
-}
-
 export type ParamsOption<T> = T extends { parameters: any }
   ? HasRequiredKeys<T["parameters"]> extends never
     ? { params?: T["parameters"] }
     : { params: T["parameters"] }
-  : never;
+  : DefaultParamsOption;
+// v7 breaking change: TODO uncomment for openapi-typescript@7 support
+// : never;
```

Comment:
This is a minor breaking change: in `openapi-typescript@7`, more structures are generated as `never`, including all `parameters`, so the inference is improved. But in v6, missing items are simply omitted. This is the only place that makes a difference, but it’s still a breaking change nonetheless and it means `openapi-fetch` will need a breaking release when the time comes.